lib/extensions.js

```diff
@@ -40,6 +40,35 @@ module.exports = function(realmConstructor) {
     setConstructorOnPrototype(realmConstructor.Results);
     setConstructorOnPrototype(realmConstructor.Object);
 
+    //Add async open API
+    Object.defineProperties(realmConstructor, getOwnPropertyDescriptors({
+        open(config) {
+            return new Promise((resolve, reject) => {
+                realmConstructor._waitForDownload(config, (error) => {
+                    if (error) {
+                        reject(error);
+                    }
+
+                    let syncedRealm = new realmConstructor(config);
+                    //FIXME: RN hangs here. Remove when node's makeCallback alternative is implemented
```

Comment:
we should really just get the makeCallback pr implemented/merged - think the work is already done?